tests/phpunit/tests/admin/includes/wpGetPluginActionButton.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * Tests for wp_get_plugin_action_button().
+ *
+ * @group plugins
+ * @group admin
+ *
+ * @covers ::wp_get_plugin_action_button
+ */
+class Tests_Admin_Includes_WpGetPluginActionButton extends WP_UnitTestCase {
+
+	/**
+	 * User role.
+	 *
+	 * @var WP_Role
+	 */
+	private static $role;
+
+	/**
+	 * User ID.
+	 *
+	 * @var int
+	 */
+	private static $user_id;
+
+	/**
+	 * Test plugin data.
+	 *
+	 * @var stdClass
+	 */
+	private static $test_plugin;
+
+	/**
+	 * Sets up properties and adds a test plugin before any tests run.
+	 */
+	public static function set_up_before_class() {
+		parent::set_up_before_class();
+
+		require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+
+		$role_name = 'wp_get_plugin_action_button-test-role';
+		add_role( $role_name, 'Test Role' );
+
+		self::$role        = get_role( $role_name );
+		self::$user_id     = self::factory()->user->create( array( 'role' => $role_name ) );
+		self::$test_plugin = (object) array(
```

Comment:
This approach works well for the specific needs of this ticket and the issue it addresses. That said, the function under test will need additional plugin mashups (i.e. different `$data`) for its various test scenarios.

It might be good to start off with a helper function that builds / creates the plugin for each test. 

Is there already examples / usage within the test suite? Yes.

[Example](https://github.com/WordPress/wordpress-develop/blob/f30f29c24f28c13f4002bd2e02dd97fc3f440e75/tests/phpunit/tests/rest-api/rest-plugins-controller.php#L1138), `WP_REST_Plugins_Controller_Test::create_test_plugin()`.

What do you think @costdev?

Reply:
The more I think about this -> let's tackle it when adding more tests. The current approach is within scope for this ticket.